wgpu-hal/src/lib.rs

```diff
@@ -700,20 +827,36 @@ pub trait Queue: WasmNotSendSync {
     ///
     /// Valid usage:
     ///
-    /// - All of the [`CommandBuffer`][cb]s were created from
-    ///   [`CommandEncoder`][ce]s that are associated with this queue.
+    /// - All of the [`CommandBuffer`][cb]s must have been created from
+    ///   [`CommandEncoder`][ce]s that were constructed from the [`Device`][d]
+    ///   associated with this [`Queue`].
+    ///
+    /// - A given [`CommandBuffer`][cb] must only be submitted for execution
+    ///   once, even if prior submissions returned errors.
     ///
-    /// - All of those [`CommandBuffer`][cb]s must remain alive until
-    ///   the submitted commands have finished execution. (Since
-    ///   command buffers must not outlive their encoders, this
-    ///   implies that the encoders must remain alive as well.)
+    /// - All of those [`CommandBuffer`][cb]s must remain alive until the
+    ///   submitted commands have finished execution. (Since command buffers
+    ///   must not outlive their encoders, this implies that the encoders must
+    ///   remain alive as well.)
     ///
-    /// - All of the [`SurfaceTexture`][st]s that the command buffers
-    ///   write to appear in the `surface_textures` argument.
+    /// - All the resources used by a submitted [`CommandBuffer`][cb]
+    ///   ([`Texture`][t]s, [`BindGroup`][bg]s, [`RenderPipeline`][rp]s, and so
+    ///   on) must remain alive until the command buffer finishes execution.
+    ///
+    /// - All of the [`SurfaceTexture`][st]s in `surface_textures` must be
+    ///   configured for use with the [`Device`][d] associated with this
+    ///   [`Queue`], typically by calling [`Surface::configure`].
+    ///
+    /// - All of the [`SurfaceTexture`][st]s that the command buffers write to
+    ///   must appear in the `surface_textures` argument.
```

Comment:
style: I think definite articles in reference to resources here is superfluous.

```suggestion
    /// - All resources used by a submitted [`CommandBuffer`][cb]
    ///   ([`Texture`][t]s, [`BindGroup`][bg]s, [`RenderPipeline`][rp]s, and so
    ///   on) must remain alive until the command buffer finishes execution.
    ///
    /// - All [`SurfaceTexture`][st]s in `surface_textures` must be
    ///   configured for use with the [`Device`][d] associated with this
    ///   [`Queue`], typically by calling [`Surface::configure`].
    ///
    /// - All [`SurfaceTexture`][st]s that `command_buffers` write to
    ///   must appear in the `surface_textures` argument.
```